ragaai_catalyst/redteaming/llm_generator.py

```
@@ -81,3 +80,51 @@ def generate_response(self, system_prompt: str, user_prompt: str, max_tokens: in
+                    content = content.split("\n", 1)[1] if content.startswith("```json") else content[3:]
+                    if "```" in content:
+                        content = content[:content.rfind("```")].strip()
+                    else:
+                        content = content.strip()
+                
+                content = json.loads(content)
+            
+            return content
+            
+        except Exception as e:
+            raise Exception(f"Error generating LLM response: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)